mk_lc.py

```diff
@@ -136,241 +178,266 @@ def lc(oid, band):
-    savepath = f'/hpc/group/cosmology/lna18/roman_sim_imgs/Roman_Rubin_Sims_2024/{oid}/{oid}_{band}_lc_coadd.csv'
-    results.write(savepath, format='csv', overwrite=True)
-
-def main(argv):
-    o_id = int(sys.argv[1])
-    # b = str(sys.argv[2])
-    lc(o_id,band)
-    
-if __name__ == '__main__':
-    main(sys.argv)
-
-# MAKE ONE LC PLOT
-# Read in truth:
-truthpath = f'/hpc/group/cosmology/phy-lsst/work/dms118/roman_imsim/filtered_data/filtered_data_{oid}.txt'
-
-truth = Table.read(truthpath, names=['mjd','filter','mag'], format='ascii')[:-1]
-
-truth['mjd'] = [row['mjd'].replace('(','') for row in truth]
-truth['mag'] = [row['mag'].replace(')','') for row in truth]
-
-truth['mjd'] = truth['mjd'].astype(float)
-truth['mag'] = truth['mag'].astype(float)
-truth['filter'] = [filt.upper() for filt in truth['filter']]
-truth = truth[truth['filter'] == band[:1]]
-truth.sort('mjd')
-
-truthfunc = interp1d(truth['mjd'], truth['mag'], bounds_error=False)
-
-# Get photometry.
-phot_path = f'/hpc/group/cosmology/lna18/roman_sim_imgs/Roman_Rubin_Sims_2024/{oid}/{oid}_{band}_lc_coadd.csv'
-phot = Table.read(phot_path)
-phot.sort('mjd')
-
-# Make residuals. 
-resids = truthfunc(phot['mjd']) - (phot['mag'] + phot['zpt'])
-
-fig, ax = plt.subplots(nrows=2,sharex=True,height_ratios=[1,0.5],figsize=(14,14))
-plt.subplots_adjust(hspace=0)
-
-ax[0].errorbar(phot['mjd'],phot['mag']+phot['zpt'],yerr=phot['magerr'],
-                marker='o',linestyle='',color='mediumpurple')
-ax[0].plot(truth['mjd'],truth['mag'],
-                marker=None,linestyle='-',color='mediumpurple')
-
-ax[1].axhline(0,color='k',linestyle='--')
-ax[1].errorbar(phot['mjd'],resids,
-                marker='o',linestyle='',color='mediumpurple')
-
-
-ax[0].set_xlim(start,end)
-ax[0].set_ylim(29,21)
-ax[0].set_ylabel(band)
-ax[1].set_xlabel('MJD')
-
-fig.suptitle(oid,y=0.91)
-plt.savefig(f'figs/{oid}_{band}_coadd.png', dpi=300, bbox_inches='tight')
-
-
-# for b in bands:
-#     lc(oid,b)
-
-# lc(oid,band)
-
-# fig = plt.figure(figsize=(14,16))
-# lcidx = [0,1,2,6,7,8,12]
-# resididx = [3,4,5,9,10,11,15]
-
-# bands = ['R062','Z087','Y106','J129','H158','F184','K213']
-# truthbands = ['R','Z','Y','J','H','F','K']
-# colors=['mediumslateblue','darkturquoise','lightgreen','mediumvioletred','darkorange','darkgrey','black']
-# cb_p = 0
-# cb_t = 0
-
-# # Read in truth:
-# truthpath = f'/hpc/group/cosmology/phy-lsst/work/dms118/roman_imsim/filtered_data/filtered_data_{oid}.txt'
-# truth = Table.read(truthpath, names=['mjd','filter','mag'], format='ascii')[:-1]
-
-# truth['mjd'] = [row['mjd'].replace('(','') for row in truth]
-# truth['mag'] = [row['mag'].replace(')','') for row in truth]
-
-# truth['mjd'] = truth['mjd'].astype(float)
-# truth['mag'] = truth['mag'].astype(float)
-# truth['filter'] = [filt.upper() for filt in truth['filter']]
-
-# Make plot:
-# gs = GridSpec(nrows=6,ncols=3, height_ratios=[1,.33,1,.33,1,.33],width_ratios=[1,1,1],wspace=0,hspace=0)
```

Comment:
I definitely want to keep this function--it makes a plot with all LCs for one SN after all bands have run. I guess there should be a different place for this, maybe in phrosty.plotting?